token-metadata/js/src/state.ts

```diff
@@ -32,12 +38,28 @@ function isNonePubkey(buffer: Uint8Array): boolean {
     return true;
 }
 
-export function unpack(buffer: Buffer): TokenMetadata {
-    const tlv = new TlvState(buffer, 8, 4);
-    const bytes = tlv.firstBytes(TOKEN_METADATA_DISCRIMINATOR);
-    if (bytes === null) {
-        throw new TokenMetadataError('Invalid Data');
-    }
+// Pack TokenMetadata into byte slab
+export const pack = (meta: TokenMetadata): Uint8Array => {
+    const encoder = getStructEncoder([
+        ['updateAuthority', getBytesEncoder({ size: 32 })],
+        ['mint', getBytesEncoder({ size: 32 })],
+        ['name', getStringEncoder()],
+        ['symbol', getStringEncoder()],
+        ['uri', getStringEncoder()],
+        ['additionalMetadata', getArrayEncoder(getTupleEncoder([getStringEncoder(), getStringEncoder()]))],
+    ]);
```

Comment:
You can save the code duplication here by instead importing `getArrayCodec`, `getBytesCodec`, etc. and define this as `codec`. Then it will yield both `encode` and `decode` from the same constant object.

Reply:
Oh i didn't know there was that functionality. That will be a lot cleaner. Changed it 👍